Add flow, types and sql files to load data into hana

yutingshe opened the pull request:
### Merge Checklist

Please cross check this list if additions / modifications needs to be done on top of your core changes and tick them off. Reviewer can as well glance through and help the developer if something is missed out.

- [ ] Automated Tests (Jasmine integration tests, Unit tests, and/or Performance tests)
- [ ] Updated Manual tests / Demo Config
- [ ] Documentation (Application guide, Admin guide, Markdown, Readme and/or Wiki)
- [ ] Verified that local development environment is working with latest changes (integrated with latest `develop` branch)
- [ ] following best practices in [code review doc](../code_review.md)

yutingshe committed:
Add flow, types and sql files to load data into hana

yutingshe committed:
Update package.json

mwaiyee committed:
allow rkernel image to build push every develop, release branch and workflow dispatch (#751)

p-hoffmann committed:
fix fs access (#767)

* fix fs access

* allow batch execution

* Fix

Signed-off-by: Peter Hoffmann <954078+p-hoffmann@users.noreply.github.com>

---------

Signed-off-by: Peter Hoffmann <954078+p-hoffmann@users.noreply.github.com>

Zhimin-arya commented on flows/base/hana_load_plugin/flow.py:
```diff
@@ -0,0 +1,121 @@
+from .types import DataloadOptions
+
+# Constants
+DATASET = "GiBleed"
+CDM_VERSION = "5.3"
+BASE_URL = f"https://github.com/OHDSI/EunomiaDatasets/raw/main/datasets/{DATASET}/{DATASET}_{CDM_VERSION}.zip"
+
+DATA_DIR = Path("flows") / "base" / "hana_ingestion" / DATASET
+ZIP_PATH = DATA_DIR / f"{DATASET}_{CDM_VERSION}.zip"
+EXTRACT_DIR = DATA_DIR / f"{DATASET}_{CDM_VERSION}"
+CREATE_SCRIPT_DIR = Path("/app/flows/hana_load_plugin/create_script")
+
+SQL_FILES_ORDER = [
+    "hana_ddl.sql",
+    "hana_primarykey.sql",
+    "hana_indices.sql"
+    # "hana_constraints.sql",
```
Can you put them in a separate file, e.g. constants.py

brandantck committed:
[Analyze-2215] Align concept search to one api (#756)

* add filters to vocabulary search and pagination support

* add score to webapi concept

* update ui to use d2e-webapi for concept search

* add endpoint to get concept set by id

* add get concepts for d2ewebapi

* inrcease body limit to 10mb

* update types

* add concept set save and update

* add endpoint to get concepts count

* update conceptset to use username instead of userid

mwaiyee committed:
[Devops-2255] Update GHA setupdemo to build UI (#743)

* test ui change

* edit gha for ui

* edit condition

* temp fix

* add free disk space

* test ui change

* indicate ui steps

* test ui mount

* add yarn

* add ci false

* add dependency

* combine build step

* fix gha

* fix gha

* revert ui test change

* update gha

* test ui and dependecy

* supress

* fix

yutingshe committed:
Add constant.py

yutingshe committed:
Update flow.py

SantanM committed:
update reticulate for DeepPLP (#769)

* update reticulate for DeepPLP

Signed-off-by: Santan Maddi <55120902+SantanM@users.noreply.github.com>

* add treatment patterns to verify installation

Signed-off-by: Santan Maddi <55120902+SantanM@users.noreply.github.com>

* update reticulate in r-kernel

Signed-off-by: Santan Maddi <55120902+SantanM@users.noreply.github.com>

---------

Signed-off-by: Santan Maddi <55120902+SantanM@users.noreply.github.com>

amit-sharma-d4l committed:
Upgrading TreatmentPatterns version for both R ohdsi kernel and Hades flow container - the older version was 3.0.2 and was not compatible with Strategus. Upgrading to 3.1.1 (#770)

yutingshe committed:
Signing commit

alicia-koh commented on flows/base/hana_load_plugin/create_script/hana_constraints.sql:
```diff
@@ -0,0 +1,158 @@
+--postgresql CDM Foreign Key Constraints for OMOP Common Data Model 5.3 3
```
change to hana

alicia-koh commented on flows/base/hana_load_plugin/create_script/hana_ddl.sql:
```diff
@@ -0,0 +1,471 @@
+--postgresql CDM DDL Specification for OMOP Common Data Model 5.3;
```
change to hana

alicia-koh commented on flows/base/hana_load_plugin/create_script/hana_indices.sql:
```diff
@@ -0,0 +1,130 @@
+/*postgresql OMOP CDM Indices
```
change to hana

alicia-koh commented on flows/base/hana_load_plugin/create_script/hana_primarykey.sql:
```diff
@@ -0,0 +1,27 @@
+--postgresql CDM Primary Key Constraints for OMOP Common Data Model 5.3 2
```
change to hana

KangJingA committed:
Update strategus for treatment patterns (#763)

* add TreatmentPatterns to DockerFile

* use correct library

* increase file limit

* Update docker-compose.yml

* update database connection

* initial template for treatment pathways

* Update yarn.lock

* update study viewer

* Update docker-compose.yml

* Update result_viewer_template.ts

* Update services.ts

* Update result_viewer_template.ts

* increase file limit at index.ts

* upgrade treatment patterns in hades

* revert docker changes

Zhimin-arya committed:
[e2e] Add 3 test cases (#491)

* e2e test for:
1. admin portal legal page
2. concepts sets in concept page and PA
3. ETL-report unitl "save report"
4. research login

* dev e2e

* 1. update concept-sets; 2. add revert-specific-version

* add data-mapping test (scan file)

* move out data-mapping e2e test

* remove redundant files

* update revert-specific-version

* comment Build docker images for e2e test

* update revert-specific-version snapshots

* update admin-portal-legal-page

* update revert-specific-version

* update concept-sets

* update concept-sets

* update concept-sets

* uncomment build docker image

yutingshe committed:
Update .sql name and download path

p-hoffmann committed:
fix sql connection error (#789)

* fix sql connection error

* fix

alicia-koh committed:
Update according to passed in data type (#774)

KangJingA committed:
Update env.ts (#782)

mwaiyee committed:
[Adhoc] Add GHA build plugin files in release branch v0.9.0 for v0.9.x patches  (#775)

* add GHA build plugin files to run on workflow_dispatch only for v0.9.0 patch

* rm triggers except workflow_dispatch

* mod workflows

---------

Co-authored-by: Peter Hoffmann <954078+p-hoffmann@users.noreply.github.com>

KangJingA committed:
fix analysis UI and jobplugin (#787)

* Update PrefectController.ts

* only add edge when selectedNodeId exists

* remove strategus output for treatment patterns node

* Update d2e-demosetup-dev.yml

* Update concept-sets.spec.ts

amit-sharma-d4l committed:
fixed a bug in the installation URLs for TreatmentPatterns (#786)

khairul-syazwan committed:
[PA] Improve notification bar (#781)

* Align colors

* Create NotificationStack, remove sap var & eventBus

* Change message type for no patient count to 'info'

* Reset notifications on next query

* Auto search on click of Gender filter card

* Remove log

* Remove toast msg reset

* Remove sap var

* Add compatConfig

* fix demosetup gha

* Remove searchOnClick

---------

Co-authored-by: mwaiyee <waiyee.man@data4life-asia.care>

alicia-koh commented on flows/base/hana_load_plugin/flow.py:
```diff
@@ -0,0 +1,116 @@
+    # create schema only when successfully unzip
+    if not dbdao.check_schema_exists(schema):
+        logger.info(f"Creating schema {schema} ...")
+        dbdao.create_schema(schema)
```
This can be moved inside the task `run_create_scripts` or be its own task so we know at which step the flow fails at. Add logger.error

alicia-koh commented on flows/base/hana_load_plugin/flow.py:
```diff
@@ -0,0 +1,116 @@
+
+
+@task(log_prints=True)
+def run_create_scripts(schema: str, dbdao: DBDao):
```
Rename to `run_create_datamodel_scripts` instead for clarity

alicia-koh commented on flows/base/hana_load_plugin/flow.py:
```diff
@@ -0,0 +1,116 @@
+        logger.info(f"Creating schema {schema} ...")
+        dbdao.create_schema(schema)
+
+    run_create_scripts(schema, dbdao)
```
Add a hook here to drop the schema if creation of tables/pk/indexes fails:
See https://github.com/OHDSI/Data2Evidence/blob/e39361ae472caaefc97bc6e0a280cda8c5754392/flows/base/omop_cdm_plugin/flow.py#L45

LSriragavan committed:
Reorganize the HTTP test files (#719)

* re-org http test files

* remove folder services/mri-db

* copy hades build fix

PR: https://github.com/OHDSI/Data2Evidence/pull/769

Signed-off-by: LSriragavan <45031616+LSriragavan@users.noreply.github.com>

---------

Signed-off-by: LSriragavan <45031616+LSriragavan@users.noreply.github.com>

yutingshe committed:
Update the flow with hook

yutingshe committed:
Wrap drop_schema_hook as a prefect task

alicia-koh committed:
Use renv.lock and uv.lock for reproducible package installation - 1 (#662)

* Create .dockerignore

* Update .gitignore

* Move drivers out of shared utils

* Update README.md

* Update dockerfiles

- Switch to duckdb v1.3.2
- Use uv for search embedding, loyalty score, cohort survival, i2b2

* add vss extension

* Update search embedding flow

* Update search embedding dockerfile

* Update base dockerfile

* Add error handling to download extension script

* Update Dockerfile

* Update Dockerfile

* Move drivers out of shared utils

* Use root user in search embedding

* Revert changes to cohort survival

* Change name of env for clarity

khairul-syazwan committed:
[PA] Start domain value search when clicking dropdown (#790)

* Search empty string on click

* Allow empty string for domain values from ref

* Revert backend changes

* Add missing args

* Add back searchQuery to map

* Switch ternary

suwarnoong committed:
[Analyze-703] app & component registry for PA navbar (#798)

* 703 app & component registry for PA navbar

* 703 add types and clean up

* Revert accidental commit for the dataset id

* 703 add isLocal return type

p-hoffmann committed:
add hana execute function (#772)

p-hoffmann committed:
speedup ext build (#831)

* speedup ext build

* fix

* fix

* fix

p-hoffmann committed:
fix llama build (#833)

Signed-off-by: Peter Hoffmann <954078+p-hoffmann@users.noreply.github.com>

alicia-koh commented on flows/_shared_flow_utils/create_dataset_tasks.py:
```diff
@@ -104,7 +104,7 @@ def create_and_assign_roles_task(dbdao: DaoBase, schema: str):
         dbdao.grant_read_privileges(schema, dbdao.read_role)
 
-
+@task(log_prints=True)
```
Remove the task decorator

alicia-koh commented on flows/base/hana_load_plugin/flow.py:
```diff
@@ -0,0 +1,124 @@
+            table_name,
+            dbdao.engine,
+            schema=schema,
+            if_exists="replace",
```
change to "append" so that the data type from the dataframe doesn't override the ones defined in the DDL

alicia-koh commented on flows/base/package.json:
```diff
@@ -19,6 +19,34 @@
+              }
+            }
+          },
+          "type": "hana_load"
```
change type to "datamodel"

alicia-koh commented on flows/base/package.json:
```diff
@@ -19,6 +19,34 @@
     "flow": {
       "image": "ghcr.io/ohdsi/d2e/flow-base",
       "flows": [
+        {
```
add an array with key "datamodels" here with "omop5-3" as the only element
Reference:
https://github.com/OHDSI/Data2Evidence/blob/e0d1e74afa5b81371819201e700f4a70489f6478/flows/base/package.json#L590C1-L593C12

yutingshe committed:
Modify flow details

p-hoffmann committed:
pgwire improvements (#830)

* adjust pgwire

* fix

* Update docker-compose.yml

Signed-off-by: Peter Hoffmann <954078+p-hoffmann@users.noreply.github.com>

* Remove 'trex-circe' and 'trex-llama' from PLUGINS_SEED

Signed-off-by: Peter Hoffmann <954078+p-hoffmann@users.noreply.github.com>

* update trex

---------

Signed-off-by: Peter Hoffmann <954078+p-hoffmann@users.noreply.github.com>

p-hoffmann committed:
refactor llama ext (#832)

* fix llama ext build

* fix

* Update build.sh

Signed-off-by: Peter Hoffmann <954078+p-hoffmann@users.noreply.github.com>

* fix

* fix

* fix

---------

Signed-off-by: Peter Hoffmann <954078+p-hoffmann@users.noreply.github.com>

p-hoffmann committed:
improve trex dev (#839)

* improve trex dev

* fix

* fix

* Update docker-compose-local.yml

Signed-off-by: Peter Hoffmann <954078+p-hoffmann@users.noreply.github.com>

---------

Signed-off-by: Peter Hoffmann <954078+p-hoffmann@users.noreply.github.com>

yutingshe committed:
Test verification
